src/DataFramesMeta.jl

```diff
@@ -392,10 +391,35 @@ function transform(g::GroupedDataFrame; kwargs...)
     idx1 = [1; 1 .+ idx2[1:end-1]]
     for (k, v) in kwargs
         first = v(g[1])
-        result[k] = Array{eltype(first)}(undef, size(result, 1))
-        result[idx1[1]:idx2[1], k] = first
-        for i in 2:length(g)
-            result[idx1[i]:idx2[i], k] = v(g[i])
+        if first isa AbstractVector 
+            if length(first) != size(g[1], 1)
```

Comment:
AFAICT this check should be done for all groups, right?

Reply:
Probably? I ommitted it because the extra check seemed slow. 

Its the same function and similar input for each group (different selections of the same vector). Unless there are some really degenerate functions which change output size erratically based on the size of the input vector, it probably isn't necessary.

Reply:
You can never know what the user-provided function will return, and we really don't want to return invalid results. Are these checks so expensive? In my tests, they have a noticeable impact, but not so large either.

Reply:
added a check